src/Selenium2Library/keywords/_screenshot.py

```diff
@@ -17,49 +17,85 @@ def __init__(self):
         but the path does not exist it will be created.
 
-        `css` can be used to modify how the screenshot is taken. By default
-        the bakground color is changed to avoid possible problems with
-        background leaking when the page layout is somehow broken.
+        New in Selenium2Library 1.8
+
+        If ``filename`` contains {index} characters, it will be automatically
+        replaced with running index. The running index is unique for each
+        different filename. The absolute path of the saved screenshot is always
+        returned and it does not depend does the ``filename`` contain {index}.
+        See example 1 and 2 for more details.
+
+        The {index} uses Python [https://docs.python.org/2/library/string.html#string-formatting|String Formatting]
+        and it is possible to format the {index} with [https://docs.python.org/2/library/string.html#grammar-token-width|width]
+        . The example 3 uses width to format the {index} to desired length.
+        The width is also the only tested and supported option for {index}.
+        Other formatting options may also work but are not tested/supported.
+
+        Example 1:
+        | ${dir1} = | Capture Page Screenshot |
+        | File Should Exist | ${OUTPUTDIR}${/}selenium-screenshot-1.png |
+        | Should Be Equal | ${dir1} | ${OUTPUTDIR}${/}selenium-screenshot-1.png |
+        | ${dir2} = | Capture Page Screenshot |
+        | File Should Exist | ${OUTPUTDIR}${/}selenium-screenshot-2.png |
+        | Should Be Equal | ${dir2} | ${OUTPUTDIR}${/}selenium-screenshot-2.png |
+
+        Example 2:
+        | ${dir1} = | Capture Page Screenshot | ${OTHER_DIR}${/}other-{index}-name.png |
+        | ${dir2} = | Capture Page Screenshot | ${OTHER_DIR}${/}some-other-name-{index}.png |
+        | ${dir3} = | Capture Page Screenshot | ${OTHER_DIR}${/}other-{index}-name.png |
+        | File Should Exist | ${OTHER_DIR}${/}other-1-name.png |
+        | Should Be Equal | ${dir1} | ${OTHER_DIR}${/}other-1-name.png |
+        | File Should Exist | ${OTHER_DIR}${/}some-other-name-1.png |
+        | Should Be Equal | ${dir2} | ${OTHER_DIR}${/}some-other-name-1.png |
+        | File Should Exist | ${OTHER_DIR}${/}other-2-name.png |
+        | Should Be Equal | ${dir3} | ${OTHER_DIR}${/}other-2-name.png |
+
+        Example 3:
+        | Capture Page Screenshot | ${OTHER_DIR}${/}sc-{index:06}.png |
+        | File Should Exist | ${OTHER_DIR}${/}sc-000001.png |
+
         """
         path, link = self._get_screenshot_paths(filename)
         self._create_directory(path)
 
         if hasattr(self._current_browser(), 'get_screenshot_as_file'):
-          if not self._current_browser().get_screenshot_as_file(path):
-              raise RuntimeError('Failed to save screenshot ' + filename)
+            if not self._current_browser().get_screenshot_as_file(path):
+                raise RuntimeError('Failed to save screenshot ' + link)
```

Comment:
I commented above that unrelated PEP-8 changes typically should be submitted separately. Indentation fix to code that is anyway touched is definitely fine. I might even remove the unnecessary empty lines around this if/else block...